### DISCUSSION TITLE
Generate `enum` based on DB enums

### DISCUSSION THREAD

**Discussion by bombillazo** [supabase/supabase#14613] [Feature Requests]
Hello, it would benefit our TS development if the Supabase type generator generates enum`s based on the Enum constructs extracted from the database.

Although I know they are not strictly types, they are commonly used in TS development. I envision it looking something like:

```ts
// generated.db.schema.ts
...
export interface Database {
  ...
  Enums: {
    my_enum:
        | 'red'
        | 'green'
        | 'blue';
  ...
}

enum my_enum {
  red = 'red',
  green = 'green',
  blue = 'blue',
}
```

**Comment by vickkhera** [comment]
Kinda sorta related, can you refer to the generated enum type in a JSDoc reference like this? Or does the import just stop at Database?

```
@param {import('../database.types.js').Database.public.Enums.user_role_type} role
```
